pkg/solana/txm/txm.go

```diff
@@ -781,6 +782,28 @@ func (txm *Txm) Enqueue(ctx context.Context, accountID string, tx *solanaGo.Tran
 		return fmt.Errorf("failed to store new transaction with ID %s: %w", id, err)
 	}
 
+	// If a dependency transaction ID is provided, handle waiting and enqueuing asynchronously.
+	if cfg.DependencyTxID != "" {
+		go func(msg pendingTx, depID string) {
+			err := txm.waitForTxStatus(ctx, depID, types.Finalized)
+			if err != nil {
+				txm.lggr.Errorw("dependency transaction did not reach desired state", "dependencyTxID", depID, "error", err)
+				err = txm.txs.OnPrebroadcastError(msg.id, txm.cfg.TxRetentionTimeout(), txmutils.FatallyErrored, TxDependencyFail)
```

Comment:
In the case of ATA transactions, we were thinking they're probably retryable. If we always mark as `FatallyErrored` here, we would never retry. Maybe it's better to move this state transition into `waitForTxStatus`? Then we can just have the dependent transaction inherit the status whether it's failed or fatal? wdyt?

Reply:
Yeah good idea, but instead of moving the state transition into `waitForTxStatus`, I think it makes sense to just return a state there so it can be handled in any way by the caller.